ipaserver/install/ca.py

```diff
@@ -57,9 +80,14 @@ def install_check(standalone, replica_config, options):
 
     realm_name = options.realm_name
     host_name = options.host_name
-    subject_base = options.subject
 
-    if replica_config is not None:
+    if replica_config is None:
+        ca_subject = options.ca_subject
+        subject_base = options.subject_base
+    else:
+        ca_subject = lookup_ca_subject(options.realm_name)
+        subject_base = replica_config.subject_base
```

Comment:
Hm, actually this does not work fine :(

I cannot set the knob value directly because, in the replica_install case, the `ca_subject` and `subject_base` knobs have been replaced by a `removed` property.  (`ipalib/install/service.py:51`).

I cannot set `options._ca_subject` (or `._subject_base`) and then read it back later because the `options.__getattr__` raises `AttributeError` if the attr name does not correspond to a knob name (`ipapython/install/core.py:571`).

I find the installer/knob code quite hard to follow... do you have a suggestion?  (I certainly agree with your original point that it is preferable to compute the values once only, and cache them).